### PR TITLE
add reactions for Insurance <=> Families

### PR DESCRIPTION
**Pull request by h4uges**
- add FamiliesToInsuranceReaction
- add InsuranceToFamiliesReaction
- add tests for FamiliesToInsuranceReaction -> OPEN: change to ViewBasedVitruvApplicationTest
- add first tests for InsuranceToFamiliesReaction -> OPEN: improve test coverage

### CODE REVIEW

**Comment by JanWittler** [comment]
Thanks for the PR. We recently removed the compound reaction triggers (`created and inserted`, `deleted and removed`, #34). Roughly checking your code, I think everything should continue working if you replace the `created and inserted` triggers with `inserted` triggers.

**Comment by h4uges** [comment]
FamiliesToInsuranceTests - testDeleteFamilyRegister() causes an stack overflow
-> discuss if this a problem in the reaction or in the way the test is written

**Comment by h4uges** [comment]
@HeikoKlare @tsaglam So jetzt könnt ihr mit dem Review beginnen ;)

File: bundles/tools.vitruv.applications.demo.insurancefamilies/src/tools/vitruv/applications/demo/insurancefamilies/insurance2families/InsuranceToFamiliesHelper.xtend [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	
+	def static void informUserAboutReplacementOfClient(UserInteractor userInteractor, InsuranceClient insuranceClient, Family oldFamily){
+		var StringBuilder message = new StringBuilder()
+			.append("Insurance Client ")
+			.append(insuranceClient.name)
+			.append(" has been replaced by another insurance client in his family (")
+			.append(oldFamily.lastName)
+			.append("). Please decide in which family and role ")
+			.append(insuranceClient.name)
+			.append(" should be.")
```
Comment (tsaglam) [inline bundles/tools.vitruv.applications.demo.insurancefamilies/src/tools/vitruv/applications/demo/insurancefamilies/insurance2families/InsuranceToFamiliesHelper.xtend:53]: You could use [template expressions](https://www.eclipse.org/xtend/documentation/203_xtend_expressions.html#templates) here if you want. Also in other instances in this class.

File: bundles/tools.vitruv.applications.demo.insurancefamilies/src/tools/vitruv/applications/demo/insurancefamilies/insurance2families/InsuranceToFamiliesHelper.xtend [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+		val StringBuilder parentOrChildMessageBuilder = new StringBuilder()
+			.append("You have renamed ")
+			.append(oldFullname)
+			.append(" to ")
+			.append(newFullname)
+			.append(", which might cause the corresponding member in the families model to change its position inside a family.")
+			.append(" Which position should this member, who was a ")
+			.append(if (wasChildBefore) "child" else "parent")
+			.append(" before, have after the renaming?")
```
Comment (tsaglam) [inline bundles/tools.vitruv.applications.demo.insurancefamilies/src/tools/vitruv/applications/demo/insurancefamilies/insurance2families/InsuranceToFamiliesHelper.xtend:113]: Also here, for example.

File: tests/tools.vitruv.applications.demo.insurancefamilies.tests/src/tools/vitruv/applications/demo/insurancefamilies/tests/insurance2families/AbstractInsuranceFamiliesTest.xtend [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+				claimInsuranceDatabase(it) => [
+					insuranceclient += createInsuranceClient[ name = fullName(FIRST_MOM_1, LAST_NAME_1) gender = Gender.FEMALE]
+				]
+			]
+			insertCount++;
+		}
+	
+		if(insertSon){
+			decideParentOrChild(PositionPreference.Child)
+			if (insertCount > 0) decideNewOrExistingFamily(FamilyPreference.Existing, 1)
```
Comment (tsaglam) [inline tests/tools.vitruv.applications.demo.insurancefamilies.tests/src/tools/vitruv/applications/demo/insurancefamilies/tests/insurance2families/AbstractInsuranceFamiliesTest.xtend:142]: inconsistent formatting: `if()` vs. `if ()`.

Comment (tsaglam) [reply tests/tools.vitruv.applications.demo.insurancefamilies.tests/src/tools/vitruv/applications/demo/insurancefamilies/tests/insurance2families/AbstractInsuranceFamiliesTest.xtend:142]: Also in lines 129 and 131.

File: tests/tools.vitruv.applications.demo.insurancefamilies.tests/src/tools/vitruv/applications/demo/insurancefamilies/tests/util/InsuranceQueryUtil.xtend [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+@Utility
+class InsuranceQueryUtil {
+	
+	static def claimInsuranceDatabase(View view){
+		view.getRootObjects(InsuranceDatabase).claimOne
+	}
+	
+	static def claimInsuranceClient(InsuranceDatabase insuranceDatabase, String firstName, String lastName){
+		insuranceDatabase.insuranceclient.filter[c | c.name == firstName + " " + lastName].claimOne
```
Comment (tsaglam) [inline tests/tools.vitruv.applications.demo.insurancefamilies.tests/src/tools/vitruv/applications/demo/insurancefamilies/tests/util/InsuranceQueryUtil.xtend:18]: In Xtend you can often shorten these statements from
`c | c.name == firstName + " " + lastName`
to
`name == firstName + " " + lastName`

**Review by tsaglam — CHANGES_REQUESTED** [review]
I found a few minor issues for you to consider.

**Review by tsaglam — APPROVED** [review]
Looks good!

**Pull request merged**